lib/index.js

```diff
@@ -22,11 +22,23 @@ var AppsFlyer = module.exports = integration('AppsFlyer')
  */
 
 AppsFlyer.ensure(function(msg, settings) {
-  // Check to ensure message has advertisingId as a property of device object
-  var deviceInfo = msg.device() || {};
-  if (!deviceInfo.advertisingId) {
-    return this.invalid('Events to AppsFlyer must specify an advertisingId as a contextual property of the device object');
-  }
+
+  /**
+  * Temporarily removing this check per: https://segment.zendesk.com/agent/tickets/73014.
+  * Despite what their documentation says, AppsFlyer's API will accept events wihtout an
+  * advertisingId / idfa property. This ensure block was causing issues for customers
+  * that have been using this integration without sending advertisingId previously.
```

Comment:
might be helpful to add that main use case is for folks sending purely server side data